bin/list-nginx-class-ingresses.rb

```diff
@@ -3,42 +3,42 @@
 # Script to build the list of ingresses and its namespaces with
 # the ingress-class annotations as "nginx" or empty i.e "" or ingress
 # with no annotations and into a json file
+# This script will list ingress for all non-prod namespaces. For production ingresses
+# change if non_production?(namespace) to if production?(namespace)
 
 require "json"
 require "open3"
 
 NGINX_CLASS_INGRESS_LIST_FILE = "nginx_class_ingresses.json"
 
 def main
-  nginx_class_ingresses = list_ingresses_has_annotations("nginx")
-  null_class_ingresses = list_ingresses_has_annotations("")
-  no_annotation_ingresses = list_ingresses_no_annotations
-  list = nginx_class_ingresses + null_class_ingresses + no_annotation_ingresses
+  $namespaces = get_namespaces
+  nginx_class_ingresses = ingresses_matching_class("nginx")
+  null_class_ingresses = ingresses_matching_class("")
+  no_annotation_ingresses = ingresses_matching_no_class
 
-  File.write(NGINX_CLASS_INGRESS_LIST_FILE, list.to_json)
+  list = nginx_class_ingresses.compact + null_class_ingresses.compact + no_annotation_ingresses.compact
+
+  puts "Total ingress listed: #{list.size}"
+
+  File.write(NGINX_CLASS_INGRESS_LIST_FILE, JSON.pretty_generate(list))
 end
 
-def list_ingresses_has_annotations(target_ingress_class)
+def ingresses_matching_class(target_ingress_class)
   ingress_array = []
-  get_ingresses.map do |ingress|
-    if ingress.dig("metadata").include?("annotations")
-      if ingress.dig("metadata", "annotations", "kubernetes.io/ingress.class").to_s == target_ingress_class
-        ingress_name = ingress.dig("metadata", "name")
-        namespace = ingress.dig("metadata", "namespace")
-        ingress_array.push({namespace: namespace, ingress_name: ingress_name})
-      end
-    end
+  get_ingresses
+    .filter { |ingress| ingress.dig("metadata", "annotations", "kubernetes.io/ingress.class").to_s == target_ingress_class }
+    .map do |ingress|
+    ingress_array.push(non_production_tuple(ingress))
   end
   ingress_array
 end
 
-def list_ingresses_no_annotations
+def ingresses_matching_no_class
   ingress_array = []
   get_ingresses.map do |ingress|
     unless ingress.dig("metadata").include?("annotations")
```

Comment:
The combination of `map` and `unless` here is a little difficult to understand.

You can use `reject` instead:
https://ruby-doc.org/core-2.6/Array.html#method-i-reject

e.g.
```
[1,2,3,4,5].reject {|i| i.even?} => [1, 3, 5]
```

So, you could do this:

```
  get_ingresses.reject { |ingress| ingress.dig("metadata").include?("annotations") }
```

Reply:
Updated  the code